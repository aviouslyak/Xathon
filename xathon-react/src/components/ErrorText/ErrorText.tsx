import React, { HTMLAttributes } from "react";

interface ErrorTextProps extends HTMLAttributes<HTMLElement> {
  text: string;
}
const ErrorText: React.FC<ErrorTextProps> = ({ text, className, ...props }) => {
  return <p className={`text-red-500 ${className ? className : ""}`}>{text}</p>;
};

export default ErrorText;
