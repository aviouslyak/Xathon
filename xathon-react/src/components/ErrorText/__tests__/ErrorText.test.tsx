import { cleanup, render, screen } from "@testing-library/react";
import ErrorText from "../ErrorText";

describe("<ErrorText />", () => {
  let component: HTMLElement;
  beforeEach(() => {
    render(<ErrorText text="foo" className="class" />);
    component = screen.getByText("foo");
  });

  test("renders text", () => {
    expect(component).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  test("renders className", () => {
    expect(component).toHaveClass("class");
  });

  test("does not have className undefined by default", () => {
    cleanup();
    render(<ErrorText text="foo" />);
    component = screen.getByText("foo");
    expect(component).not.toHaveClass("undefined");
  });
});
