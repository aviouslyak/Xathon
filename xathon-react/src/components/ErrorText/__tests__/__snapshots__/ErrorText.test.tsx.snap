// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText /> matches snapshot 1`] = `
<p
  class="text-red-500 class"
>
  foo
</p>
`;
